NonSucking.Framework.Serialization/NoosonGenerator.cs

```diff
@@ -755,9 +771,72 @@ var typeParamsWithConstraints
             if (methodType == MethodType.DeserializeWithCtor)
                 retType = new(typeSymbol.ToDisplayString(), "", new(), "The deserialized instance.");
 
+            string desName = Consts.Deserialize;
+            if (methodType == MethodType.DeserializeSelf)
+                desName = Consts.DeserializeSelf;
+
```

Comment:
```suggestion
            string desName = methodType == MethodType.DeserializeSelf ? Consts.DeserializeSelf : Consts.Deserialize;

```

Reply:
I implemented it this ways, because one might want other names in the future as well, but the chance for this to happen might be quite low or non existing